File: 027 Remove Elements.py
import random
'''
27 Remove Elements
https://oj.leetcode.com/problems/remove-elements/
'''
# generate a random int array with limits [a,b]
def createRandList(n,a,b):
    A = []
    for i in range(n):
        A.append(random.randint(a,b))
    return A

def removeElements(array, tag):
    i = 0
    j = 0
    while i < len(array):
        while i < len(array) and array[i] == tag: 
            i += 1
        if i < len(array):
            array[j] = array[i]
            i += 1
            j += 1
    return array[0:j]
    
def main():
    array = createRandList(20,1,6)
    print array
    print removeElements(array,2)

if __name__ == "__main__":
    main()

File: 027 Remove Elements.js
/**
 * Remove Element
 * https://leetcode.com/problems/remove-element/
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val)
{
    var j = 0; // return length
    for(var i = 0; i < nums.length; i++)
    {
        if(nums[i] != val)
        {
            nums[j] = nums[i];
            j++;
        }
    }

    nums.length = j;
    return j;
};
